cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/mazha/CouchClient.java

```diff
@@ -108,9 +109,10 @@ private ExecuteResult(InputStream stream,
                 case 3:
                     // 3xx redirection
                     throw new CouchException("Unexpected redirection (3xx) code encountered", responseCode);
-                case 4:
+                case 4: case 5:
```

Comment:
I still don't like the logic here. Breaking out 1xx,2xx,3xx,4xx,5xx is the most logical way to partition things because they are distinct ranges of error codes with different means. It doesn't make sense to effectively say "if 4 or 5" and then later say "oh but only if 5". I think it's more logical to have a case statement breaking them down to the hundreds and then refining within each hundred.